src/workloads/execution/Densify.yml

```diff
@@ -0,0 +1,421 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload test tests the performance of the $densify stage. $densify can operate on both dates
+  and numbers.
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: &batchSize 1000
+    Document:
+      partitionKey: {^RandomInt: {min: 1, max: 100}}
+      # Test a subset of units that have different characteristics.
+      number: {^RandomInt: {min: 0, max: 20000}} # Seperate codepath for numeric DensifyValues.
+      milliseconds: {^RandomDate: {min: "2021-01-01T00:00:00.000Z", max: "2021-01-01T00:00:20.000Z"}} # Most basic date unit.
+      hours: {^RandomDate: {min: "2021-01-01", max: "2023-01-01"}} # Date unit with constant number of milliseconds.
+      months: {^RandomDate: {min: "2021-01-01", max: "4000-01-01"}} # Date unit with non-constant number of milliseconds.
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+
+# $densify stages always add a $sort on the field that is being densified, so this phase adds 
+# indexes on the date/number fields so that we aren't performing an in-memory sort.
+- Name: CreateIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        createIndexes: Collection0
+        indexes: 
+        - key: {number: 1}
+          name: number
+        - key: {milliseconds: 1}
+          name: milliseconds
+        - key: {hours: 1}
+          name: hours
+        - key: {months: 1}
+          name: months
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+- Name: Densify
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 10
+    Database: *db
+    Operations:
+
+    # Metrics are grouped by the step unit they operate with (numiric, millisecond, hour, month).
+    # Each group has a metric for each possible codepath (Partition/no partition, explicit/full/partitioned bounds).
+
+    # Numeric metrics. 
+    - OperationMetricsName: NumericFullDensify
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{
+          $densify: {
+            field: "number", 
+            range: {
+              bounds: "full", 
+              step: {^RandomInt: {min: 1, max: 11}}
+            }
+          }
+        }]
+        cursor: {batchSize: *batchSize}
```

Comment:
Unfortunately with the way that Genny workloads currently work, the results from this operation as it's written now will only measure how much work it would take to get the first batch of results, and not the performance of the whole operation. And because the batchSize here is significantly smaller than the number of docs in the collection (and even smaller than the amount of documents we would eventually want to output) this could give us a false understanding of performance of the $densify stage as a whole.

There are several ways to get the performance results of the whole operation (like wrapping this command in an explain and using `executionStats`) but the one that's most consistent and least likely to couple the performance of these workloads with non-$densify code is to try and fit all the results into the first batch size. Could you make a back-of-the-envelope calculation of the size of each document, as well the expected number of documents we would generate and return, and adjust the size of the docs and the batch size accordingly? Happy to talk about other methods as well if you want to explore another avenue for how to do this!

Reply:
Modifying the batch size makes sense to me -- will do!

Reply:
Increased batch size to 30k, since we're unlikely to generate more documents than that and since each document will only take up 5 bytes of space, this will be way less than 1mb.